firebase-installations/src/main/java/com/google/firebase/installations/remote/FirebaseInstallationServiceClient.java

```diff
@@ -177,20 +180,27 @@ public InstallationResponse createFirebaseInstallation(
   private void writeFIDCreateRequestBodyToOutputStream(
       HttpURLConnection httpURLConnection, @NonNull String fid, @NonNull String appId)
       throws IOException {
-    OutputStream outputStream = httpURLConnection.getOutputStream();
-    if (outputStream == null) {
-      throw new IOException(
-          "Cannot send CreateInstallation request to FIS. No OutputStream available.");
-    }
-
-    GZIPOutputStream gzipOutputStream = new GZIPOutputStream(outputStream);
     try {
-      gzipOutputStream.write(
-          buildCreateFirebaseInstallationRequestBody(fid, appId).toString().getBytes("UTF-8"));
+      writeRequestBodyToOutputStream(
+          httpURLConnection, getJsonBytes(buildCreateFirebaseInstallationRequestBody(fid, appId)));
     } catch (JSONException e) {
       throw new IllegalStateException(e);
-    } finally {
-      gzipOutputStream.close();
+    }
+  }
+
+  private static byte[] getJsonBytes(JSONObject jsonObject) throws UnsupportedEncodingException {
+    return jsonObject.toString().getBytes("UTF-8");
+  }
+
+  private static void writeRequestBodyToOutputStream(URLConnection urlConnection, byte[] jsonBytes)
+      throws IOException {
+    try (OutputStream outputStream = urlConnection.getOutputStream()) {
```

Comment:
Probably this is not an issue, but I just want to make sure that we are not running into an issue here:
If outputStream is NULL this will not cause an error when Java is trying to close the stream, right?
Did you test this?